operator/internal/status/lokistack.go

```diff
@@ -149,53 +159,18 @@ func checkForZoneawareNodes(ctx context.Context, k client.Client, zones []lokiv1
 	return true, true, nil
 }
 
-func updateCondition(ctx context.Context, k k8s.Client, req ctrl.Request, condition metav1.Condition) error {
-	var stack lokiv1.LokiStack
-	if err := k.Get(ctx, req.NamespacedName, &stack); err != nil {
-		if apierrors.IsNotFound(err) {
-			return nil
+func generateWarnings(ctx context.Context, cs *lokiv1.LokiStackComponentStatus, k k8s.Client, req ctrl.Request, stack *lokiv1.LokiStack) ([]metav1.Condition, error) {
+	warnings := make([]metav1.Condition, 0)
+	for _, sc := range stack.Status.Storage.Schemas {
+		if sc.Version != lokiv1.ObjectStorageSchemaV13 {
+			warnings = append(warnings, metav1.Condition{
```

Comment:
See the comment on the warning message: I think we should change this to only check if the schema configuration ends in a schema that uses the latest version.